diff --git a/python/tvm/driver/tvmc/compiler.py b/python/tvm/driver/tvmc/compiler.py
index 98cc5ff7d..675aa21b4 100644
--- a/python/tvm/driver/tvmc/compiler.py
+++ b/python/tvm/driver/tvmc/compiler.py
@@ -45,12 +45,8 @@ from .workspace_pools import generate_workspace_pools_args, workspace_pools_reco
 logger = logging.getLogger("TVMC")
 
 
-@register_parser
-def add_compile_parser(subparsers, _, json_params):
-    """Include parser for 'compile' subcommand"""
-
-    parser = subparsers.add_parser("compile", help="compile a model.")
-    parser.set_defaults(func=drive_compile)
+def add_compile_arguments(parser, json_params):
+    """Generate arguments for compile parser"""
     parser.add_argument(
         "--cross-compiler",
         default="",
@@ -150,6 +146,15 @@ def add_compile_parser(subparsers, _, json_params):
 
     generate_workspace_pools_args(parser)
 
+@register_parser
+def add_compile_parser(subparsers, _, json_params):
+    """Include parser for 'compile' subcommand"""
+
+    parser = subparsers.add_parser("compile", help="compile a model.")
+    parser.set_defaults(func=drive_compile)
+    add_compile_arguments(parser, json_params)
+
+
 
 def drive_compile(args):
     """Invoke tvmc.compiler module with command line arguments
